spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -649,6 +668,39 @@ private String deleteOrcCompressionStrategy() {
         .option(SparkWriteOptions.COMPRESSION_STRATEGY)
         .sessionConf(SparkSQLProperties.COMPRESSION_STRATEGY)
         .tableProperty(DELETE_ORC_COMPRESSION_STRATEGY)
-        .parseOptional();
+        .defaultValue(orcCompressionStrategy())
+        .parse();
+  }
+
+  private long dataAdvisoryPartitionSize() {
+    long defaultValue =
+        advisoryPartitionSize(DATA_FILE_SIZE, dataFileFormat(), dataCompressionCodec());
+    return advisoryPartitionSize(defaultValue);
+  }
+
+  private long deleteAdvisoryPartitionSize() {
+    long defaultValue =
+        advisoryPartitionSize(DELETE_FILE_SIZE, deleteFileFormat(), deleteCompressionCodec());
+    return advisoryPartitionSize(defaultValue);
+  }
+
+  private long advisoryPartitionSize(long defaultValue) {
+    return confParser
```

Comment:
I think this is okay, although since the default differs between data and deletes, it seems strange not to allow setting this specifically for deletes. The write option makes sense, but the table and session properties would likely differ.

Reply:
Yeah, I was thinking about this too.

Reply:
I think we typically use a variation that adds `merge`, `update`, or `delete` somewhere. I don't feel strongly so let's go without this for now. We can add more settings later.